src/lava/lib/dl/netx/hdf5.py

```diff
@@ -0,0 +1,458 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier:  BSD-3-Clause
+
+"""HDF5 network exchange module."""
+
+from typing import List, Tuple, Union
+from lava.magma.core.decorator import implements
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+import numpy as np
+import h5py
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import InPort, OutPort
+from lava.proc.lif.process import LIF
+from lava.proc.sdn.process import Sigma, Delta, SigmaDelta
+from lava.lib.dl.netx.utils import NetDict
+from lava.lib.dl.netx.utils import optimize_weight_bits
+from lava.lib.dl.netx.blocks.process import Input, Dense, Conv
+from lava.lib.dl.netx.blocks.models import AbstractPyBlockModel
+
+
+class Network(AbstractProcess):
+    """Generates a Lava process for the network described in hdf5 config.
+
+    Parameters
+    ----------
+    net_config : str
+        name of the hdf5 config file.
+    num_layers : int, optional
+        number of blocks to generate. An integer valuew will only generate the
+        first ``num_layers`` blocks in the description. The actual number of
+        generated layers may be less than ``num_layers``. If it is None, all
+        the layers are generated. Defaults to None.
+    has_graded_input : bool, optional
+        flag indicating the input spike type of input layer. Defaults to False.
+    """
+    def __init__(
+        self,
+        **kwargs: Union[int, Tuple[int, ...]]
+    ) -> None:
+        super().__init__(**kwargs)
+        self.filename = kwargs.pop('net_config')
```

Comment:
Is there a difference between net_config and filename?

Reply:
They are same